pkg/build/controller/build/build_controller.go

```diff
@@ -1208,25 +1208,65 @@ func (bc *BuildController) createBuildPod(build *buildv1.Build) (*buildUpdate, e
 		return update, nil
 	}
 
-	klog.V(4).Infof("Pod %s/%s for build %s is about to be created", build.Namespace, buildPod.Name, buildDesc(build))
-	pod, err := bc.podClient.Pods(build.Namespace).Create(buildPod)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		// Log an event if the pod is not created (most likely due to quota denial).
-		bc.recorder.Eventf(build, corev1.EventTypeWarning, "FailedCreate", "Error creating build pod: %v", err)
-		update.setReason(buildv1.StatusReasonCannotCreateBuildPod)
-		update.setMessage("Failed creating build pod.")
-		return update, fmt.Errorf("failed to create build pod: %v", err)
+	existingPod, err := bc.podClient.Pods(build.Namespace).Get(
+		buildPod.Name, metav1.GetOptions{},
+	)
+	if err != nil && !errors.IsNotFound(err) {
+		update.setReason(buildv1.StatusReasonError)
+		update.setMessage("Failed getting build pod.")
+		return update, fmt.Errorf("failed to get build pod: %v", err)
+	} else if errors.IsNotFound(err) {
+		// Before starting a new pod we need to make sure the namespace
+		// doesnt contain pre filled in ConfigMaps as these may contain
+		// invalid content.
+		if tainted, err := bc.taintedNamespace(build); err != nil {
+			update.setReason(buildv1.StatusReasonError)
+			update.setMessage("Failed verifying namespace.")
```

Comment:
We'll need a clearer error message - such as "Unexpected build configuration data found."

Reply:
The error message here should say something like: "I was trying to look for unexpected build configuration data but I have failed to do so, will try your build again later". Not sure how to express that properly.

"Failed looking for unexpected build configuration" ?